### ISSUE TITLE
Incomplete docs for "authenticate" function

### ISSUE THREAD

**Issue by reilem** [fastify/fastify-passport#30]
## 🐛 Bug Report

(Sorry if this is not the correct issue type)

The `authenticate` function has an optional callback parameter, but the signature is not provided. Maybe I am not looking carefully, but the docs say "The callback has the following signature", but I can't seem to see the signature anywhere.

I had to look in the source code to find out what was going wrong, and only then realised that the user is given as 4th parameter in the callback and is not available in the request object.

Just adding something like this to the docs would've made my life a lot easier 30 minutes ago 😅 
```
Callback: (request, reply, err, user?, info?, status | statuses?) => Promise<void>
```

Let me know if there is any way I can help.

**Comment by mcollina** [comment]
Would you like to send a Pull Request to add this to the README?

**Comment by reilem** [comment]
Sure, I'll see if I have time sometime later today

**Comment by reilem** [comment]
Sorry for the delay, work was busy but I finally got round to it. 😅 See #35

**Comment by airhorns** [comment]
Closed by #35
